cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -443,6 +444,13 @@ runTxBuild (AnyCardanoEra era) (AnyConsensusModeParams cModeParams) networkId mS
       (utxo, pparams, eraHistory, systemStart, stakePools) <-
         newExceptT . fmap (join . first ShelleyTxCmdAcquireFailure) $
           executeLocalStateQueryExpr localNodeConnInfo Nothing $ \_ntcVersion -> runExceptT $ do
+            unless (null txinsc) $ do
+              collateralUtxo <- firstExceptT ShelleyTxCmdTxSubmitErrorEraMismatch . newExceptT . queryExpr
+                $ QueryInEra eInMode
+                $ QueryInShelleyBasedEra sbe (QueryUTxO . QueryUTxOByTxIn $ Set.fromList txinsc)
+              txinsExist txinsc collateralUtxo
+              notScriptLockedTxIns collateralUtxo
```

Comment:
This is the same as https://github.com/input-output-hk/cardano-node/blob/d8f81b9dd789adc7659e400322a5cf85f3e78f3b/cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs#L462 except adapted to work in the monadic query expression.